### DISCUSSION TITLE
`list[Union[Type1, Type2]]` raises `UserWarning`

### DISCUSSION THREAD

**Discussion by GuiEpi** [pydantic/pydantic#10553] [Question]
Hello,

Pydantic raises `UserWarning: Pydantic serializer warnings` when my `list` contains only one element in the `Union`. However, in the documentation I read that: 
- unions require **only one member** to be valid.

I hesitate to disable the warnings but I don't know if it's a good practice.

I also have the same problem with `bool` when it receives 0 or 1 whereas in the documentation I read that:
- The integers `0` or `1`

and specifies that you can use `StrictBool` to validate only `True` and `False`

---
**My main question is, am I doing something wrong or should I disable the warnings?**

I thank anyone who can help me. Don't hesitate to ask me if you need any further information!

**Comment by contactnana** [comment]
which python version are you using, if you are on Python 3.10+, use the | operator for when something could be one of a few types
x: list[int | str] = [3, 5, "test", "fun"]  
or x: list[type1 | type 2 ]

**Reply by GuiEpi** [reply]
Hello and thank you for your answer, but `Union` and `|` are identical. I tried anyway but it always returns the warning.

**Reply by GuiEpi** [reply]
I've also tried `Union[list[Type1], list[Type2]]` but now I have this waning `Union[list[Type2], list[Type2]]` but got `list`
